test/kubernetes/e2e/features/services/tcproute/suite.go

```diff
@@ -81,14 +88,22 @@ func (s *testingSuite) TestConfigureTCPRouteBackingDestinationsWithMultiServices
 
 	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, multiListenerGatewayAndClientManifest)
 	s.Assert().NoError(err, "can apply gateway and client manifest")
+	s.testInstallation.Assertions.EventuallyGatewayCondition(s.ctx, "tcp-gateway", "default", v1.GatewayConditionAccepted, metav1.ConditionTrue, timeout)
 
 	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, multiBackendServiceManifest)
 	s.Assert().NoError(err, "can apply backend service manifest")
+	s.testInstallation.Assertions.EventuallyObjectsExist(s.ctx, proxyService, proxyDeployment)
 
 	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, multiTcpRouteManifest)
 	s.Assert().NoError(err, "can apply tcproute manifest")
+	s.testInstallation.Assertions.EventuallyTCPRouteCondition(s.ctx, "tcp-app-1", "default", v1.RouteConditionAccepted, metav1.ConditionTrue, timeout)
```

Comment:
nit: Do you think it would be valuable to assert on a list of conditions?

Reply:
I was considering ^ but I decided to start simple, mainly b/c we need to check the condition status for the different types, e.g. `Accecpted=True` while `Conflicted=False`.